lib/agentx.c

```diff
@@ -223,6 +225,7 @@ DEFUN (agentx_enable,
 		events = list_new();
 		agentx_events_update();
 		agentx_enabled = 1;
+		hook_call(agentx_enabled);
 	}
```

Comment:
what does hook call has to do with a mib implementation.
at least this should be separate commit. ( because other daemons have snmp and do not need to implement this hook).
at best, this should be a separate pull request.

Reply:
Hi, 

Yes, I can split this hook call into a separate pull request.

The purpose of the hook call is used to trigger the ldpe and lde processes to register for specific MIB oids.

As you probably know, LDP runs as 3 processes: ldpd, ldpe, lde.
The main ldpd process is an FRR client.
The other 2 LDP processes are not FRR clients.

The MIBs are implemented in LDP processes: ldpe, lde.

With this code change:
- The LDP main process registers for the agentx_enable hook.  
- When the agentx_enable hook runs, the LDP main process sends an 'agentx enabled' message to the ldpe and lde processes.
- When ldpe and lde processes receive this 'agentx enabled' message, they call REGISTER_MIB to register for specific MIB oids.

Reply:
@pguibert6WIND 
Just FYI, this hook was merged in as a separate PR: https://github.com/FRRouting/frr/pull/8023

@pguibert6WIND 
Just FYI, I created PR https://github.com/FRRouting/frr/pull/8147 for LDP SNMP without statistics.  I will mark this PR 7944 as a draft.